app/scripts/move-popup/dimItemTag.directive.js

```diff
@@ -12,24 +12,22 @@
     `
   });
 
-  ItemTagController.$inject = ['$rootScope', 'SyncService', 'dimPlatformService', 'dimSettingsService', 'dimItemTagService'];
+  ItemTagController.$inject = ['$scope', '$rootScope', 'dimSettingsService'];
 
-  function ItemTagController($rootScope, SyncService, dimPlatformService, dimSettingsService, dimItemTagService) {
+  function ItemTagController($scope, $rootScope, dimSettingsService) {
     var vm = this;
 
     vm.settings = dimSettingsService;
-    vm.selected = _.find(vm.settings.itemTags, function(tag) {
-      return tag.type === vm.item.tag;
+    $scope.$watch('vm.item.dimInfo.tag', function() {
```

Comment:
More curious than anything else... but instead of a watcher and then explicitly calling save, could we instead do something like:

```
var _tag;
Object.defineProperty(vm.item.dimInfo, 'tag', function() {
  get: function() {
    return _tag;
  },
  set: function(val) {
    _tag = val;
    vm.selected = _.find(vm.settings.itemTags, function(tag) {
      return tag.type === val;
    });
    vm.item.dimInfo.save();
  }
});
```

or would that not really help things here?